pkg/tracker/apitracker.go

```diff
@@ -0,0 +1,285 @@
+	}
+
+	a.portStorage.add(containerID, successfullyForwarded)
+
+	if len(errs) != 0 {
+		return fmt.Errorf("%w: %+v", ErrExposeAPI, errs)
+	}
+
+	return nil
+}
+
+// Get looks up the port mapping by containerID and returns the result.
+func (a *APITracker) Get(containerID string) nat.PortMap {
+	return a.portStorage.get(containerID)
+}
+
+// Remove a single entry from the port storage and calls the
+// /services/forwarder/unexpose endpoint to remove the forwarded the port mappings.
+func (a *APITracker) Remove(containerID string) error {
+	portMap := a.portStorage.get(containerID)
+	defer a.portStorage.remove(containerID)
+
+	var errs []error
+
+	for _, portBindings := range portMap {
+		for _, portBinding := range portBindings {
+			// The unexpose API only supports IPv4
+			ipv4, err := isIPv4(portBinding.HostIP)
+			if !ipv4 || err != nil {
+				continue
+			}
+
+			log.Debugf("calling %s API for the following port binding: %+v", unexposeAPI, portBinding)
+
+			err = a.unExpose(
+				&types.UnexposeRequest{
+					Local: ipPortBuilder(portBinding.HostIP, portBinding.HostPort),
+				})
+			if err != nil {
+				errs = append(errs,
+					fmt.Errorf("unexposing %+v failed: %w", portBinding, err))
+
+				continue
+			}
+		}
+	}
+
+	if len(errs) != 0 {
+		return fmt.Errorf("%w: %+v", ErrUnexposeAPI, errs)
+	}
+
+	return nil
+}
+
+// RemoveAll calls the /services/forwarder/unexpose
+// and removes all the port bindings from the tracker.
+func (a *APITracker) RemoveAll() error {
+	var errs []error
+
+	allPortMappings := a.portStorage.getAll()
+
+	for _, portMapping := range allPortMappings {
+		for _, portBindings := range portMapping {
+			for _, portBinding := range portBindings {
+				// The unexpose API only supports IPv4
+				ipv4, err := isIPv4(portBinding.HostIP)
+				if !ipv4 || err != nil {
+					continue
+				}
+
+				log.Debugf("calling %s API for the following port binding: %+v", unexposeAPI, portBinding)
+
+				err = a.unExpose(
+					&types.UnexposeRequest{
+						Local: ipPortBuilder(portBinding.HostIP, portBinding.HostPort),
+					})
+				if err != nil {
+					errs = append(errs,
+						fmt.Errorf("RemoveAll unexposing %+v failed: %w", portBinding, err))
+
+					continue
+				}
+			}
+		}
+	}
+
+	a.portStorage.removeAll()
+
+	if len(errs) != 0 {
+		return fmt.Errorf("%w: %+v", ErrUnexposeAPI, errs)
+	}
+
+	return nil
+}
+
+func (a *APITracker) expose(exposeReq *types.ExposeRequest) error {
+	bin, err := json.Marshal(exposeReq)
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("sending a HTTP POST to %s API with expose request: %v", exposeAPI, exposeReq)
+	req, err := http.NewRequestWithContext(
+		context.Background(),
+		http.MethodPost,
+		a.urlBuilder(exposeAPI),
+		bytes.NewReader(bin))
+	if err != nil {
+		return err
+	}
+
+	res, err := a.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	return verifyResposeBody(res)
+}
+
+func (a *APITracker) unExpose(unexposeReq *types.UnexposeRequest) error {
```

Comment:
The case here is odd.
```suggestion
func (a *APITracker) unexpose(unexposeReq *types.UnexposeRequest) error {
```